src/verifast1.ml

```diff
@@ -1885,8 +1897,16 @@ module VerifyProgram1(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
                      end
         ) attrs;
         ctxt#assert_term (ctxt#mk_le s max_uintptr_term);
-        ctxt#assert_term (ctxt#mk_lt (ctxt#mk_intlit 0) s);
-        let rec iter1 fmap fds has_ghost_fields =
+        let lte, bases_size = 
+          match bases with
+          | [] -> (* no bases, a struct cannot have size 0 -> lt *)
+                  ctxt#mk_lt, int_zero_term
+          | _ ->  (* only match on non-virtual bases because virtual bases are not supported yet -> throws unmatched case otherwise *)
+                  (* compiler may introduce padding in bases and the derived struct's members may fit in this padding -> le *)
+                  ctxt#mk_le, bases |> List.fold_left (fun size (CxxBaseSpec (l, base, false)) -> struct_size_partial smapwith0 l base |> ctxt#mk_add size) int_zero_term
+        in
+        lte bases_size s |> ctxt#assert_term;

```

Comment:
Are you sure? If the compiler can put fields of the derived struct inside the padding of base structs, can it not also put one base inside the padding of another base? For example, consider
```c++
struct A { int x; short y; } // 2 bytes of padding at the end
struct B { char c; }
struct C : A, B {}
```
Could it be that `sizeof(struct A) == 8` and `sizeof(struct C) == 8`?

Bottom line: how useful is it to assert lower bounds on the size of a derived struct? What use cases do you envisage  that would need this information? Perhaps do not bother for now?

Reply:
Removed the assertions